utils/managed_bytes.cc

```diff
@@ -171,8 +178,15 @@ void managed_bytes_view_base::remove_prefix(size_t n) {
 
 managed_bytes_view
 managed_bytes_view::substr(size_t offset, size_t len) const {
```

Comment:
We should have a special case for no `len` (take all). Maybe default `len` to `-1`.

Reply:
sounds good, maybe define managed_bytes_view::npos as -1.

Reply:
Yes.